File: api/internal/model/player.go
package model

type Player struct {
	ID         string `json:"id"`
	Name       string `json:"name"`
	Position   string `json:"position"` // GK, DEF, MID, FWD
	Nationality string `json:"nationality"`
	Age        int    `json:"age"`
	TeamID     string `json:"team_id"`

	// Attributes (0–100)
	Pace       int `json:"pace"`
	Shooting   int `json:"shooting"`
	Passing    int `json:"passing"`
	Dribbling  int `json:"dribbling"`
	Defending  int `json:"defending"`
	Physical   int `json:"physical"`

	// Derived
	Overall    int `json:"overall"`
	Value      int64 `json:"value"` // in credits
}

File: mobile/types/player.ts
export type Position = 'GK' | 'DEF' | 'MID' | 'FWD';

export interface Player {
  id: string;
  name: string;
  position: Position;
  nationality: string;
  age: number;
  overall: number;
  pace: number;
  shooting: number;
  passing: number;
  dribbling: number;
  defending: number;
  physical: number;
  value: number;
}
